packages/markdown/remark/src/rehype-collect-headings.ts

```diff
@@ -57,3 +71,58 @@ export function rehypeHeadingIds(): ReturnType<RehypePlugin> {
 function isMDXFile(file: MarkdownVFile) {
 	return Boolean(file.history[0]?.endsWith('.mdx'));
 }
+
+/**
+ * Check if an ESTree entry is `frontmatter.*.VARIABLE`.
+ * If it is, return the variable path (i.e. `["*", ..., "VARIABLE"]`) minus the `frontmatter` prefix.
+ */
+function getMdxFrontmatterVariablePath(node: MdxTextExpression): string[] | Error {
+	if (!node.data?.estree || node.data.estree.body.length !== 1) return new Error();
+
+	const statement = node.data.estree.body[0];
+
+	// Check for "[ANYTHING].[ANYTHING]".
+	if (statement.type !== 'ExpressionStatement' || statement.expression.type !== 'MemberExpression')
```

Comment:
```suggestion
	if (statement?.type !== 'ExpressionStatement' || statement.expression.type !== 'MemberExpression')
```

Nit: check `statement` isn't null